pkgs/tools/security/zmap/0001-The-byacc-binary-in-nixpkgs.byacc-is-named-yacc.patch

```diff
@@ -0,0 +1,26 @@
+From f4a787f574287c3ee021b4fbc2220f7324dd8cdf Mon Sep 17 00:00:00 2001
+From: Maximilian Bosch <maximilian@mbosch.me>
+Date: Mon, 6 May 2019 15:25:42 +0200
+Subject: [PATCH] The byacc binary in nixpkgs.byacc is named `yacc`
+
+Nix-specific patch to get the cmake process running.
+---
+ src/CMakeLists.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 99054c6..5b36e8d 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -192,7 +192,7 @@ add_custom_command(OUTPUT parser.c
+ )
+ else()
+ add_custom_command(OUTPUT parser.c
+-    COMMAND byacc -d -o "${CMAKE_CURRENT_BINARY_DIR}/parser.c" "${CMAKE_CURRENT_SOURCE_DIR}/parser.y"
++    COMMAND yacc -d -o "${CMAKE_CURRENT_BINARY_DIR}/parser.c" "${CMAKE_CURRENT_SOURCE_DIR}/parser.y"
```

Comment:
Isn't it an overkill to create a .patch file just for ```s/byacc/yacc/``` ?
Maybe just
```postPatch = "substituteInPlace CMakeLists.txt --replace byacc yacc"```;

Reply:
I originally added the patch as there was some more stuff to do that I managed to fix without the patch. I don't bother too much about small patches which is why I kept it.
But if everybody else agress, I can fix that :)

Reply:
I'm strong supporter here of a patch because of what kind of failure it would give as compared to `substituteInPlace` where it may build without the substitution. A patch will just fail to apply.

Reply:
Ok, but

* patch applications often fail because of changes in the patched files which have been captured into ```.patch```-file being irrelevant to the intended fix.
Here we have to change a single word but captured ~15 words to track them
* if one more ```byacc``` appeared in the next version, patch won't fix it

Reply:
Arch seems to do `--program-transform-name='s,^,b,'` as a `configureFlag`, Seems we should just do that since the executable probably should be called `byacc` anyway. (it is the packages name)
See also what fedora does https://src.fedoraproject.org/rpms/byacc/blob/master/f/byacc.spec#_37

##### train of thought is doing a substitution at a lower level :smile: 

@Ma27 does this sound correct to you?

Reply:
That's indeed a good idea. I'll have a look at this :)

Reply:
It seems as this is something that only works with `./configure` scripts generated by `m4`, but not by CMake. I guess it's easier to use `substituteInPlace` then.